### ISSUE TITLE
Draft.js no longer maintained

### ISSUE THREAD

**Issue by katzmo** [springload/draftail#456]
Draft.js has not been supported for a while and will be fully archived by the end of 2022. See [Draft.js README](https://github.com/facebook/draft-js/blob/main/README.md) and https://github.com/facebook/draft-js/pull/3136 for more information.

Meta has stared working on a new extensible JavaScript web text-editor framework called [Lexical](https://github.com/facebook/lexical). Migration guides from Draft.js to Lexical is yet to come (https://github.com/facebook/lexical/issues/1641) and the framework is still in "experimental" stage.

What does that mean for Draftail? Are there any plans to fork Draft.js or switch to Lexical?

**Comment by thibaudcolas** [comment]
Hey @katzmo 👋 good question.

Draftail is used in a relatively large CMS called [Wagtail](https://wagtail.org/), which I work on as part of my day job. As long as the CMS uses Draftail, I’ll keep maintaining the editor, even if Draft.js itself is no longer maintained. We also do occasional feature development to improve how the editor works (currently working on UX improvements in an upcoming v2.0.0 release).

So – Draftail isn’t going away any time soon. There are known issues with it (#138), some due to the editor itself, some due to Draft.js. They don’t prevent the editor from being useful in a real-world setting, though obviously with rough edges in specific circumstances. Technically Draftail still uses an old version of Draft.js from 2018, well before they stopped maintenance, so we’d still have quite a bit of changes from that project to catch up to.

---

I’ll update the Draftail website and the project README to mention the state of Draft.js as this is super important info for anyone considering the editor.

As far as Lexical – personally I found Facebook’s stewardship of Draft.js as a project to be very poor. The software was great, so I have hopes about Lexical being even better, but I won’t consider it as an option until it’s proving to be battle-tested outside Facebook. Right now for example the documentation isn’t good enough in my opinion for a project of that complexity.

**Comment by dustinblanchard** [comment]
Would it make sense to explore moving to quill or something else for Wagtail?

**Comment by thibaudcolas** [comment]
@dustinblanchard certainly. There are more decent options out there these days compared to when Wagtail made the switch to Draft.js/Draftail (when IE11 support was make-or-break and browsers were much more inconsistent with how much they supported rich text editing).

There’s been some discussions about this over 2022. No plans to switch to anything just yet but the work to make Wagtail compatible with multiple editors has already been done. There’d be more work needed to create an editor-agnostic way to create rich text editor customisations (at least for simple ones). Then / around the same time, we’d review different options, and most likely implement one as a third-party package before eventually moving it in core, and then having it be the default option.

I’m hoping we’ll get to wait until further browser standardisation ([W3C Web Editing WG](https://www.w3.org/2021/06/web-editing-wg-charter.html)) before we need a replacement, but we’ll see.

**Comment by katzmo** [comment]
@thibaudcolas Thank you for sharing your thoughts and considerations for the way forward! 

I'm sure mentioning some of this also on the website and README will be very helpful, not least to show the project is still maintained despite the state of draft.js.

**Comment by minusf** [comment]
Hello. Could you please update the README?  It's a bit hidden here in the issue that project is archived. Thank you.

**Comment by lb-** [comment]
If there's ever a discussion about a library. I'd recommend looking into TipTap. It's got a solid amount of momentum, with a large community base and is very customisable.

https://tiptap.dev/

Which itself is built on ProseMirror https://tiptap.dev/guide/prosemirror

I'm not sure if these would replace Draftail or sit just under it to provide the basic framework for rich text. It's also got built in support for Code editing, and table editing. It would be a lot of work to change to this though.

**Comment by SalahAdDin** [comment]
Any update on this?

> If there's ever a discussion about a library. I'd recommend looking into TipTap. It's got a solid amount of momentum, with a large community base and is very customisable.
> 
> https://tiptap.dev/
> 
> Which itself is built on ProseMirror https://tiptap.dev/guide/prosemirror
> 
> I'm not sure if these would replace Draftail or sit just under it to provide the basic framework for rich text. It's also got built in support for Code editing, and table editing. It would be a lot of work to change to this though.

I couldn't find any Tiptap package for wagtail.

**Comment by thibaudcolas** [comment]
No update aside from [New repo organization - same maintainers #462](https://github.com/wagtail/draftail/issues/462). Draftail  still works plenty well enough there’s no hurry to build a replacement. Both TipTap and ProseMirror are great alternatives if/when someone has the time to build a Wagtail integration (or if you have the money [a feature sponsorship](https://wagtail.org/sponsor/) will definitely help make this happen!).

I’ve published a [Draftail project roadmap](https://github.com/wagtail/draftail/blob/main/ROADMAP.md) to make it clear what types of things I’d like to do with the project, which is mostly maintenance and bug fixes but also the occasional feature development.
